tests/virt/node/general/utils.py

```diff
@@ -0,0 +1,40 @@
+import logging
+import shlex
+from datetime import datetime
+
+from pyhelper_utils.exceptions import CommandExecFailed
+from pyhelper_utils.shell import run_ssh_commands
+
+from utilities.virt import running_vm
+
+LOGGER = logging.getLogger(__name__)
+MARKER_PATH = "~/reset_marker"
+
+
+def get_vm_latest_boot_time(vm):
+    result = run_ssh_commands(host=vm.ssh_exec, commands=[shlex.split("last -F reboot -n 1")])[0].strip()
+    parts = result.split()
+    date_str = " ".join(parts[4:9])
+    latest_boot_time = datetime.strptime(date_str, "%a %b %d %H:%M:%S %Y")
+    LOGGER.info(f"Latest boot time found: {latest_boot_time}")
+    return latest_boot_time
+
+
+def create_ephemeral_marker(vm):
+    run_ssh_commands(host=vm.ssh_exec, commands=[shlex.split(f"touch {MARKER_PATH}")])
+    LOGGER.info(f"Ephemeral marker created at {MARKER_PATH}")
+
+
+def verify_ephemeral_marker_exists(vm):
+    try:
+        run_ssh_commands(host=vm.ssh_exec, commands=[shlex.split(f"test -f {MARKER_PATH}")])
+        return True
+    except CommandExecFailed:
+        LOGGER.error(f"Marker file not found: {MARKER_PATH}")
+        return False
+
+
+def reset_vm_and_wait(vm):
+    LOGGER.info(f"Resetting VM {vm.name}")
+    vm.vmi.reset()
+    running_vm(vm=vm)
```

Comment:
the function running_vm is trying to start VM, in your case you need to be sure no any "start" commands were sent. I think you need to use `wait_for_running_vm` function